### PR TITLE
nx-text-link fixes RSC-734, RSC-744, RSC-749, RSC-750

### PR DESCRIPTION
**Pull request by glenhunter**
https://issues.sonatype.org/browse/RSC-750
https://issues.sonatype.org/browse/RSC-744
https://issues.sonatype.org/browse/RSC-749
https://issues.sonatype.org/browse/RSC-734

This PR fixes several bugs and a couple of design changes:
1. `nx-text-link`'s should be underlined
2. focused `nx-text-link`'s should use `outline` rather than `border`
3. fixes a horizontal scrollbar that appeared in dropdown menus with `nx-text-link`'s in them 
4. fixes the `NxIconDropdown` visual test for the open menu which wasn't working correctly and so caused the above issue to be missed

### CODE REVIEW

File: lib/src/base-styles/_nx-globals.scss [hunk 1]
```diff
@@ -15,11 +15,10 @@ $-nx-scrollable-default-height: 400px;
 .nx-text-link {
   @include nx-container-helpers.container-horizontal;
 
-  border: 1px solid transparent;
   color: var(--nx-color-link);
   cursor: pointer;
-  margin: 0 calc(0px - var(--nx-spacing-base));
-  padding: 0 calc(var(--nx-spacing-base) - 1px);
+  margin: 0;
+  padding: 0;
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-globals.scss:21]: Is it necessary to make these explicit?

File: lib/src/base-styles/_nx-globals.scss [hunk 2]
```diff
@@ -31,9 +30,8 @@ $-nx-scrollable-default-height: 400px;
   }
 
   &:focus {
-    border-color: var(--nx-color-interactive-border-focus);
     border-radius: var(--nx-border-radius);
-    outline: 0;
+    outline: 1px solid var(--nx-color-interactive-border-focus);
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-globals.scss:34]: I think `outline-offset` is needed to get the outline to the desired space from the text.

File: lib/src/components/NxDropdown/NxDropdown.scss [hunk 3]
```diff
@@ -117,6 +117,11 @@
 
 .nx-dropdown-link {
   color: var(--nx-color-link);
+  margin: 0;
+
+  &:focus {
+    border-radius: 0;
```
Comment (rpokorny) [inline lib/src/components/NxDropdown/NxDropdown.scss:123]: Is there a reason this isn't with the rest of the focus styles up around line 87?

File: lib/src/components/NxDropdown/NxDropdown.scss [hunk 4]
```diff
@@ -117,6 +117,11 @@
 
 .nx-dropdown-link {
   color: var(--nx-color-link);
+  margin: 0;
```
Comment (rpokorny) [inline lib/src/components/NxDropdown/NxDropdown.scss:120]: Seems redundant

File: gallery/visualtests/nxIconDropdown.js [hunk 5]
```diff
@@ -40,7 +40,7 @@ describe('NxIconDropdown', function() {
       await targetElement.moveTo({ xOffset: -10, yOffset: -10 });
 
       const { x, y } = await targetElement.getLocation();
-      const region = new Region(parseInt(x, 10), parseInt(y, 10), 251, 376);
+      const region = new Region(parseInt(x, 10) - 208, parseInt(y, 10), 251, 376);
```
[the remark this answers is not in the document]
Comment (glenhunter) [reply gallery/visualtests/nxIconDropdown.js]: > Minor: This seems to be a taller screenshot than necessary

Made it 30px shorter here: https://github.com/sonatype/sonatype-react-shared-components/pull/529/commits/f51aa349cc469985f30f9abfb77ae609668befb7

**Comment by glenhunter** [comment]
> I don't understand this part of the mockup. What is the right edge of the 2px pink zone aligned to? @alexalyons126
> image

I think @alexalyons126 measured as if the 24px "box" from other icon implementations was present. The space looked the same as our standard `16px` of spacing so I left it as that.

File: lib/src/base-styles/_nx-globals.scss (hunk 2 above)
Comment (glenhunter) [reply lib/src/base-styles/_nx-globals.scss:32]: > > Ah, I forgot to post the screenshots. 4px caused it to run over the adjacent text. 2px looked correct.
> 
> Are there adjustments that need to me made to the mockups to match that then?

I measured screenshots and it's `16px` in the RSC and `15px` in Sketch. Measuring from the underline to the edge of the icon. So near enough.

**Pull request merged**